addons/packages/antrea/1.2.3/test/antrea_test.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2021 VMware Tanzu Community Edition contributors. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package antrea_test
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/yaml"
+
+	"github.com/vmware-tanzu/community-edition/addons/packages/test/pkg/repo"
+	"github.com/vmware-tanzu/community-edition/addons/packages/test/pkg/ytt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Antrea Ytt Templates", func() {
+	var (
+		filePaths []string
+		values    string
+		output    string
+		err       error
+
+		configDir             = filepath.Join(repo.RootDir(), "addons/packages/antrea/1.2.3/bundle/config")
+		fileAntreaYaml        = filepath.Join(configDir, "upstream/antrea.yaml")
+		fileAntreaOverlayYaml = filepath.Join(configDir, "overlay/antrea_overlay.yaml")
+		fileValuesYaml        = filepath.Join(configDir, "values.yaml")
+		fileValuesStar        = filepath.Join(configDir, "values.star")
+	)
+	BeforeEach(func() {
+		values = ""
+	})
+
+	JustBeforeEach(func() {
+		filePaths = []string{fileAntreaYaml, fileAntreaOverlayYaml, fileValuesYaml, fileValuesStar}
+		output, err = ytt.RenderYTTTemplate(ytt.CommandOptions{}, filePaths, strings.NewReader(values))
+	})
+
+	Context("default configuration", func() {
+		It("renders a ConfigMap with a default ipam configuration", func() {
+			Expect(err).NotTo(HaveOccurred())
+			configMap := parseConfigMap(output)
+			Expect(configMap.Data["antrea-agent.conf"]).To(MatchYAML(`---
+featureGates:
+  AntreaProxy: true
+  EndpointSlice: false
+  Traceflow: true
+  NodePortLocal: false
+  AntreaPolicy: true
+  FlowExporter: false
+  NetworkPolicyStats: false
+  Egress: false
+trafficEncapMode: encap
+noSNAT: false
+serviceCIDR: 10.96.0.0/12
```

Comment:
what about adding `serviceCIDRv6: null` in this list, so we match the `values.yaml` content

Reply:
The current behavior of other fields is to not display them if they are null. Such as the `defaultMTU`.

```
#@ if/end values.antrea.config.defaultMTU:
defaultMTU: #@ values.antrea.config.defaultMTU
```
[source](https://github.com/tylerschultz/community-edition/blob/5736884161ab0fd127b9adac5524380b8c0fd5e0/addons/packages/antrea/1.2.3/bundle/config/overlay/antrea_overlay.yaml#L86)